test/integration/node-specific/mongo_client.test.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+const { setupDatabase } = require('../shared');
+const f = require('util').format;
+
+class CustomPromise extends Promise {}
+CustomPromise.prototype.isCustomMongo = true;
+
+describe('MongoClient', function () {
```

Comment:
There's another describe block that uses this same title, currently it prints out as a warning, cus I think when I added the code there was many duplicates but since then we've cleaned them up. We have to keep each title unique or the xunit file won't output every test. 

Maybe you can change this line to a throw? or just fix this name for now
https://github.com/mongodb/node-mongodb-native/blob/main/test/tools/reporter/mongodb_reporter.js#L179

Reply:
My vote is to do both; as far as renaming, just something like "MongoClient integration" would work fine

Reply:
When I tried to modify the reporter to throw instead of warn, I noticed that the afterEach session checker hooks were what failed.  For now, I'd rather just change the name of the test to fix the duplicate test warning and we can look at this later